sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -1036,22 +1037,45 @@ def predict_log_proba(self, X):
         y_prob = self.predict_proba(X)
         return np.log(y_prob, out=y_prob)
 
-    def predict_proba(self, X):
+    def predict_proba(self, X, max_row_chunk=None):
         """Probability estimates.
 
         Parameters
         ----------
         X : {array-like, sparse matrix}, shape (n_samples, n_features)
             The input data.
 
+        max_row_chunk: integer
+            The number of rows to use at a single predict call. If None optimal 
+            row number fit that fits the working_memory is used.
+
         Returns
         -------
         y_prob : array-like, shape (n_samples, n_classes)
             The predicted probability of the sample for each class in the
             model, where classes are ordered as they are in `self.classes_`.
         """
         check_is_fitted(self, "coefs_")
-        y_pred = self._predict(X)
+        # split and collate depending on the working_memory
+        x_axis0_size = X.shape[0]
+        if not max_row_chunk:
+            chunk_size = get_chunk_n_rows(row_bytes=sys.getsizeof(X[0:1]),
+                                          max_n_rows=x_axis0_size)
```

Comment:
The chunk size is not properly calculated I think.

As far as I understand, the issue is not with the size of `X` (which is already stored in memory anyway), but with how much memory is required to run predict of MLP on `X` with `n_samples, n_features`. At the very least I suspect it depends on `hidden_layer_sizes`, see the discussion in https://github.com/amueller/introduction_to_ml_with_python/issues/67#issue-282550528. We would need to find some approximate way to estimate the memory usage from the input parameters, in order to estimate the chunk_size that would respect `working_memory`.

Reply:
Then, for now, we can go with the default parameters of `10000` samples, as in the other PR you mentioned. And with the help of some data collection over various sample size, we might be able to map the parameter of chunk size to the `hyperparameter` of the `MLPClassifier`.